components/cdc/src/errors.rs

```diff
@@ -65,8 +71,8 @@ impl Error {
             ))))
             | Error::Txn(TxnError(box TxnErrorInner::Mvcc(MvccError(
                 box MvccErrorInner::Engine(EngineError(box EngineErrorInner::Request(e))),
-            ))))
-            | Error::Request(e) => e,
+            )))) => e,
+            Error::Request(e) => e.as_ref().clone(),
```

Comment:
How about `| Error::Request(box e) => e`?

Reply:
`e` is `Box<ErrorHeader>` not `ErrorHeader`